sympy/logic/boolalg.py

```diff
@@ -2390,7 +2390,7 @@ def SOPform(variables, minterms, dontcares=None):
     .. [2] https://en.wikipedia.org/wiki/Don%27t-care_term
 
     """
-    if not minterms:
+    if not len(minterms):
```

Comment:
Why not just move this check later after `minterms = _input_to_binlist(minterms, variables)`?
I think that boolean magic `not minterms` is very common way in Python to check empty lists, so I'm not happy that we should change this syntax, without any sufficient comments or documentations why we are using that
(I'm not sure about the news if it is antipattern nowadays or not)
(I also note that we have lots of people in the community, and a lot people can arbitrary change the code to `if not minterms` without understanding the context that it has problems with numpy.)

Reply:
Your suggestion worked perfectly and thanks for the tips about antipatterns and docs , I'll keep those things in mind from now while writing new code . I have done the required changed in the new commit . I'm sorry for the mess created in this PR , I made a mistake while updating the local repo and didn't realise it until much later.